compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPUVirtualVectorLowering.cpp

```diff
@@ -53,6 +56,7 @@ void LLVMCPUVirtualVectorLoweringPass::runOnOperation() {
           .setVectorTransferSplit(vectorTransferSplit);
 
   RewritePatternSet patterns(ctx);
+  arm_neon::populateLowerContractionToSMMLAPatternPatterns(patterns);
```

Comment:
I think this should only be applied when we target arm_neon. We can provide options to the pass. The start point should be below: we look at target attributes, and determine if we enable a set of patterns. Then we add options to LLVMCPUVirtualVectorLowering, and set the option based on LLVMCPUPipelineOptions. LLVMCPUVectorTransposeLowering pass is an example.

https://github.com/openxla/iree/blob/503edab02257a42e865d08fb3d7d06263c0b62a3/compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPULowerExecutableTarget.cpp#L96-L106

Reply:
I believe arm_neon is specified with a "+neon" feature as part of the target flags.  I can check for it similar to "+i8mm". I've added to the PR would that satisfy? Not clear if we need to create a separate PipeLine options for this.

@dcaballe i8mm can only be used with neon? I couldn't find a definitive answer in the documentation, but in my case I've only tested it on neon. Outside of CPUMaterializeEncodings, there are no later checks on neon specifically before adding the patterns.

Reply:
`+neon` doesn't include i8mm. You may want to add a `hasI8mmFeature` utility that checks for `+i8mm`. I'm not sure if there is a specific i8mm feature for SVE. @banach-space should know but we may want to enable it for SVE as well if that's not the case.

Reply:
Updated `LLVMCPUVirtualVectorLoweringPass` to use the pipeline options. these patterns are now gated on the `+neon` cpu features.

Reply:
Sorry if my previous comment wasn't clear: we should check for `+i8mm`, not `+neon`.

Reply:
> I'm not sure if there is a specific i8mm feature for SVE.

It's the same name as for NEON :) Btw, nice to see this happening @KoolJBlack  🙏🏻

Reply:
switch to i8mm